tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -274,6 +280,17 @@ ZTEST_USER(userspace, test_disable_mmu_mpu)
 
 		addr += 0x20000000;
 	}
+#endif
+
+#if defined(CONFIG_XTENSA_MPU)
+	/* Simply clear out all foreground MPU entries and
+	 * let the background take over.
```

Comment:
I guess I don't understand what this is testing?  It's using a hardware instruction to clear registers in the MPU hardware to expose the underlying background map defined in the hardware.

This isn't actually testing any Zephyr code, so why bother?

I mean, I get that it's testing a "disable" feature, but we don't provide one to test!  (Also if we did, it couldn't look like this.  mt8195's DSP for example has a useless uncached background map.

Reply:
The test itself is to make sure user threads cannot disable MMU/MPU. If there is such an attempt, CPU should raise a exception, and the thread killed.